repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/attributemappingvalidation/AttributeMappingPolicyValidationFilterTest.groovy

```diff
@@ -167,7 +166,105 @@ class AttributeMappingPolicyValidationFilterTest extends ReposeValveTest {
             |        roles: "{D}"
             |        expire: "{D}"
             |  version: RAX-1
-            """.stripMargin()
+            |'''.stripMargin()
+
+        when:
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: "PUT", headers: ["content-type": TEXT_YAML], requestBody: body)
+
+        then:
+        mc.receivedResponse.code == "200"
+        mc.handlings.size() == 1
+        yaml.load(body) == yaml.load(mc.handlings[0].request.body as String)
+    }
+
+    def "should validate XML with multi-value roles"() {
+        given:
+        String body =
+            '''<?xml version="1.0" encoding="UTF-8"?>
+            |<mapping xmlns="http://docs.rackspace.com/identity/api/ext/MappingRules"
+            |         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            |         xmlns:xs="http://www.w3.org/2001/XMLSchema"
+            |         xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion"
+            |         version="RAX-1">
+            |  <rules>
+            |    <rule>
+            |      <local>
+            |        <user>
+            |          <domain value="{D}"/>
+            |          <name value="{D}"/>
+            |          <email value="{D}"/>
+            |          <roles value="{Ats(http:\\/\\/schemas.xmlsoap.org\\/claims\\/Group)}"
+            |                 multiValue="true"
```

Comment:
Is this attribute specified on both the policy and the assertion?

Reply:
Yep. It sure is.